recipes/opencv/4.x/conanfile.py

```diff
@@ -0,0 +1,311 @@
+                cmake_target = component["target"]
+                lib_name = get_lib_name(component["lib"])
+                requires = component["requires"]
+                self.cpp_info.components[conan_component].names["cmake_find_package"] = cmake_target
+                self.cpp_info.components[conan_component].names["cmake_find_package_multi"] = cmake_target
+                self.cpp_info.components[conan_component].libs = [lib_name]
+                self.cpp_info.components[conan_component].includedirs = ['include/opencv4']
+                self.cpp_info.components[conan_component].requires = requires
+                if self.settings.os == "Linux":
+                    self.cpp_info.components[conan_component].system_libs = ["dl", "m", "pthread", "rt"]
+
+                if conan_component == 'opencv_highgui':
+                    self.cpp_info.components[conan_component].system_libs = ["gtkmm-2.4"]
+
+                # CMake components names
+                conan_component_alias = conan_component + "_alias"
+                cmake_component = component["lib"]
+                self.cpp_info.components[conan_component_alias].names["cmake_find_package"] = cmake_component
+                self.cpp_info.components[conan_component_alias].names["cmake_find_package_multi"] = cmake_component
+                self.cpp_info.components[conan_component_alias].requires = [conan_component]
+                self.cpp_info.components[conan_component_alias].includedirs = ['include/opencv4']
+                self.cpp_info.components[conan_component_alias].libdirs = []
+                self.cpp_info.components[conan_component_alias].resdirs = []
+                self.cpp_info.components[conan_component_alias].bindirs = []
+                self.cpp_info.components[conan_component_alias].frameworkdirs = []
+
+        def imageformats_deps():
+            components = []
+            if self.options.with_jasper:
+                components.append("jasper::jasper")
+            if self.options.with_png:
+                components.append("libpng::libpng")
+            if self.options.with_jpeg:
+                components.append("libjpeg::libjpeg")
+            if self.options.with_tiff:
+                components.append("libtiff::libtiff")
+            if self.options.with_openexr:
+                components.append("openexr::openexr")
+            return components
+
+        def eigen():
+            return ["eigen::eigen"] if self.options.with_eigen else []
+
+        def tbb():
+            return ["tbb::tbb"] if self.options.with_tbb else []
+
+        self.cpp_info.filenames["cmake_find_package"] = "OpenCV"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "OpenCV"
+
+        add_components([
+            {"target": "opencv_core",       "lib": "core",       "requires": ["zlib::zlib"] + tbb()},
+            {"target": "opencv_flann",      "lib": "flann",      "requires": ["opencv_core"] + tbb()},
```

Comment:
if core requires tbb does it need to be repeated everywhere?

Reply:
Also a good point! I also took this literally from the 2.4.13.7 conanfile. I assumed they would have a good reason to do this. I am not sure how I can test this and be sure it can go (the test_package isn't using tbb).

Reply:
Yes, it's listed for all targets. I've commented on 2.4.13.7 PR, but you can check all cmake target files here: https://gist.github.com/uilianries/e5de90e8df47b0bd6eae25a25e334c5c